docs/howto/custom_json_encoder_decoder.rst

```diff
@@ -0,0 +1,49 @@
+Define custom JSON encoders and decoders
+----------------------------------------
+
+When calling ``mytask.defer()`` to create a job, the task arguments are serialized into
+a JSON string for storing into the Postgres database.
+
+And after fetching a job from the database a Procrastinate worker needs to deserialize
+the task arguments before calling the task.
+
+By default Procrastinate relies on the JSON dumps and loads functions used by psycopg2.
+See the `psycopg2 documentation`_. But Procrastinate makes it possible to specify custom
+encoders and decoders.
+
+Here is an example involving serializing/deserializing ``datetime`` objects::
+
+    import functools
+    import json
+
+    from procrastinate import App, PostgresJobStore
+
+    # Function used for encoding datetime objects
+    def encode(obj):
+        if isinstance(obj, datetime.datetime):
+            return obj.isoformat()
+        raise TypeError()
+
+
+    # Function used for decoding datetime objects
+    def decode(dict_):
+        if "dt" in dict_:
+            dict_["dt"] = datetime.datetime.fromisoformat(dict_["dt"])
+        return dict_
+
+    json_dumps = functools.partial(json.dumps, default=encode)
+    json_loads = functools.partial(json.loads, object_hook=decode)
+
+    app = App(job_store=PostgresJobStore(json_dumps=json_dumps, json_loads=json_loads))
+
+In this example the custom JSON dumps and loads functions are based on the standard
+``json`` module's ``dumps`` and ``loads`` functions, with specific ``default`` and
+``object_hook`` for serializing and deserializing ``datetime`` objects, respectively.
+See the `Python 3 json documentation`_ for more detail.
+
+This mechanism even makes it possible to use a different JSON implemation than ``json``,
+such as `UltraJSON`_ for example.
```

Comment:
"If your encoding function starts resembling a long list of `if isisntance` calls, you may want to have a look at `functools.singledispatch` (link) for a cleaner way."

Reply:
Done.